src/js/tracker.js

```diff
@@ -738,12 +745,54 @@
 		 */
 		function addCommonContexts(userContexts) {
 			var combinedContexts = commonContexts.concat(userContexts || []);
+
+			// Add PerformanceTiming Context
 			if (autoContexts.performanceTiming) {
 				var performanceTimingContext = getPerformanceTimingContext();
 				if (performanceTimingContext) {
 					combinedContexts.push(performanceTimingContext);
 				}
 			}
+
+			// Add Optimizely Contexts
+			if (window['optimizely']) {
+
+				if (autoContexts.optimizelyExperiments) {
+					combinedContexts = combinedContexts.concat(getOptimizelyExperimentContexts());
+				}
+
+				if (autoContexts.optimizelyStates) {
+					combinedContexts = combinedContexts.concat(getOptimizelyStateContexts());
+				}
+
+				if (autoContexts.optimizelyVariations) {
+					combinedContexts = combinedContexts.concat(getOptimizelyVariationContexts());
+				}
+
+				if (autoContexts.optimizelyVisitor) {
+					var optimizelyVisitorContext = getOptimizelyVisitorContext();
+					if (optimizelyVisitorContext) {
+						combinedContexts.push(optimizelyVisitorContext);
+					}
+				}
+
+				if (autoContexts.optimizelyAudiences) {
+					combinedContexts = combinedContexts.concat(getOptimizelyAudienceContexts());
+				}
+
+				if (autoContexts.optimizelyDimensions) {
+					combinedContexts = combinedContexts.concat(getOptimizelyDimensionContexts());
+				}
```

Comment:
Can we change these `concat`s to `push`es? (The reason I use concat at the top of this function is to create a new object rather than modifying an existing one, which can cause problems when there are multiple trackers on the page sharing state.)